src/model/id.rs

```diff
@@ -5,6 +5,22 @@ use std::num::{NonZeroI64, NonZeroU64};
 
 use super::Timestamp;
 
+pub struct IDFromStrError;
+
+impl fmt::Debug for IDFromStrError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "IDFromStrError(id of zero)")
+    }
+}
+
+impl fmt::Display for IDFromStrError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Attempted to construct a id of zero.")
+    }
+}
```

Comment:
```suggestion
#[derive(Debug, Clone)]
pub struct IDFromStrError;

impl fmt::Display for IDFromStrError {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        f.write_str("attempted to construct an ID of zero")
    }
}
```

Otherwise, this is okay to merge.